File: backend/service/search/src/main/java/com/leopoldhsing/digitalhippo/search/service/ProductSearchingService.java
package com.leopoldhsing.digitalhippo.search.service;

import com.leopoldhsing.digitalhippo.model.dto.SearchingResultIndexDto;
import com.leopoldhsing.digitalhippo.model.vo.ProductSearchingConditionVo;

public interface ProductSearchingService {
    SearchingResultIndexDto searchProducts(ProductSearchingConditionVo condition);
}

File: backend/service/product/src/main/kotlin/com/leopoldhsing/digitalhippo/product/service/ProductSearchingService.kt
package com.leopoldhsing.digitalhippo.product.service

import com.leopoldhsing.digitalhippo.model.dto.SearchingResultIndexDto
import com.leopoldhsing.digitalhippo.model.vo.ProductSearchingConditionVo

interface ProductSearchingService {
    fun searchProducts(condition: ProductSearchingConditionVo): SearchingResultIndexDto
}